cardano-api/src/Cardano/Api/TextView.hs

```diff
@@ -57,8 +56,8 @@ data TextView = TextView
   } deriving (Eq, Show)
 
 instance ToJSON TextView where
-  toJSON (TextView (TextViewType tvType) (TextViewDescription desc) rawCBOR) =
-    object [ "type" .= Text.decodeUtf8 tvType
```

Comment:
The above actually produces a compiler warning in the absence of `NamedFieldPuns`.  Seems like `NamedFieldPuns` suppresses shadow variable warnings.  If we value the shadow variable warning, it's a good reason to avoid this extension - especially since we actually don't use the substance of this extension.